server/sonar-server/src/main/java/org/sonar/server/component/index/ComponentIndex.java

```diff
@@ -95,39 +116,50 @@ private static FiltersAggregationBuilder createAggregation(ComponentIndexQuery q
   }
 
   private static TopHitsBuilder createSubAggregation(ComponentIndexQuery query) {
-    TopHitsBuilder sub = AggregationBuilders.topHits(DOCS_AGGREGATION_NAME);
+    TopHitsBuilder sub = AggregationBuilders.topHits(DOCS_AGGREGATION_NAME)
+      .setHighlighterEncoder("html")
+      .setHighlighterPreTags("<mark>")
+      .setHighlighterPostTags("</mark>")
+      .addHighlightedField(createHighlighter());
     query.getLimit().ifPresent(sub::setSize);
     return sub.setFetchSource(false);
   }
 
-  private QueryBuilder createQuery(ComponentIndexQuery query, ComponentTextSearchFeature... features) {
+  private QueryBuilder createQuery(ComponentIndexQuery query, Set<String> favorites, ComponentTextSearchFeature... features) {
     BoolQueryBuilder esQuery = boolQuery();
     esQuery.filter(authorizationTypeSupport.createQueryFilter());
     ComponentTextSearchQuery componentTextSearchQuery = ComponentTextSearchQuery.builder()
       .setQueryText(query.getQuery())
       .setFieldKey(FIELD_KEY)
       .setFieldName(FIELD_NAME)
+      .setRecentlyBrowsed(query.getRecentlyBrowsed())
+      .setFavorites(favorites)
       .build();
     return esQuery.must(ComponentTextSearchQueryFactory.createQuery(componentTextSearchQuery, features));
   }
 
-  private static List<ComponentsPerQualifier> aggregationsToQualifiers(SearchResponse response) {
+  private Set<String> getFavoriteKeys() {
```

Comment:
I suggest to return UUIDs instead of keys. They are shorter and can also boost the score when matching the field _id.